nmmo/core/env.py

```diff
@@ -357,62 +361,47 @@ def _compute_scripted_agent_actions(self, actions: Dict[int, Dict[str, Dict[str,
     return actions
 
   def _compute_observations(self):
-    '''Neural MMO Observation API
-
-    Args:
-        agents: List of agents to return observations for. If None, returns
-        observations for all agents
-
-    Returns:
-        obs: Dictionary of observations for each agent
-        obs[agent_id] = {
-          "Entity": [e1, e2, ...],
-          "Task": [encoded_task],
-          "Tile": [t1, t2, ...],
-          "Inventory": [i1, i2, ...],
-          "Market": [m1, m2, ...],
-          "ActionTargets": {
-              "Attack": [a1, a2, ...],
-              "Sell": [s1, s2, ...],
-              "Buy": [b1, b2, ...],
-              "Move": [m1, m2, ...],
-          }
-        '''
-
+    '''Create an Observation object for each agent in self.agents'''
     obs = {}
+    market = Item.Query.for_sale(self.realm.datastore) # the same for all agents
 
-    market = Item.Query.for_sale(self.realm.datastore)
+    # dummy obs
```

Comment:
can we allocate these once per process? 

@functools.lru_cache(maxsize=None)
def _dummy_obs(self):
  Observation(....)

Reply:
I changed it to create the dummy obs at reset and use it during episode by copying the Observation instance and replacing the tick and agent id.